### PR TITLE
Captcha 예측 기능을 포함한 FastAPI 백엔드 구축

### PR DESCRIPTION
**Pull request by RaonJena99**
- CORS 및 세션 미들웨어 설정
- /captcha 엔드포인트로 캡차 문제 생성 및 저장
- /predict 엔드포인트로 캡차 이미지 추론 처리
- MLflow에서 등록된 PyTorch 모델 불러와 예측 수행
- HybridCNN 모델 클래스 정의
- image_tensor에 .to(device) 추가하여 모델과 디바이스 일치시킴

### CODE REVIEW

**Comment by RaonJena99** [comment]
이미지 전처리 개선 및 입력 크기 일관성 유지
- 글자 중심 정렬, padding 적용, 28x28 resize하여 모델 학습과 동일한 입력 포맷 유지
- resize된 이미지를 저장하여 디버깅 및 성능 분석에 용이하도록 구성

File: src/mlruns/0/meta.yaml
Comment (ljy2855) [inline src/mlruns/0/meta.yaml:1]: mlflow를 로컬에 두지 않고 remote로 접근해서 사용할 예정이라 해당 mlruns는 지워주세요!

File: src/models/models.py
Comment (ljy2855) [inline src/models/models.py:1]: 마찬가지로 model도 mlflow로부터 load하여 사용할 예정이라, 해당 코드도 작동 확인되셧으면 지워주세요!

File: src/routers/captcha.py [hunk 1]
```diff
@@ -1,15 +1,70 @@
...
 from schemas.captcha import CaptchaRequest, CaptchaResponse
 from utils.id_gen import generate_captcha_id
+from PIL import Image
+from torchvision import transforms
+from models.models import HybridCNN
+import mlflow, mlflow.pytorch, base64, io, torch, os, traceback
 
 router = APIRouter()
 
+# 디바이스 설정
```
Comment (ljy2855) [inline src/routers/captcha.py]: router폴더에는 로직 작성보다 endpoint 관리를 위한 파일이라 device 세팅은 models 모듈에 utils 파일에 작성해주세요

File: src/routers/captcha.py [hunk 2]
```diff
@@ -1,15 +1,70 @@
...
+
+# 모델 불러오기 및 디바이스 이동
+model = mlflow.pytorch.load_model("models:/HybridCNN@production")
+model.to(device)
+model.eval()
+
 # 메모리 기반 문제 저장소 (임시용)
 captcha_store = {}
 
+# base64 이미지 데이터 중심 정렬 전처리
```
Comment (ljy2855) [inline src/routers/captcha.py]: 마찬가지로 이미지 전처리 로직도 utils에 작성해주세요

**Review by ljy2855 — CHANGES_REQUESTED** [review]
어려우실 수 있으실텐데, 잘 해주셔서 감사합니다 :)

backend 서비스다보니 ml 모델에 대한 dependency만 제거해주시면 좋을 것 같아요!
[모델 서빙](https://github.com/ljy2855/stock_prediction/blob/main/model_serve/main.py) 이 파일을 한번 보시고 수정만 부탁드려요!

**Comment by RaonJena99** [comment]
수정을 했는데, 제대로 작동하는지는 확인하지 못했습니다.

File: src/routers/captcha.py [hunk 3]
```diff
@@ -1,108 +1,56 @@
...
-# 디바이스 설정
-device = torch.device("mps" if torch.backends.mps.is_available() else "cpu")
-
-mlflow.set_tracking_uri("http://localhost:5001") # mlflow 주소
-
-# 모델 불러오기 및 디바이스 이동
-model = mlflow.pytorch.load_model("models:/HybridCNN@production")
-model.to(device)
-model.eval()
+# 원격 ML 서비스 URL
```
Comment (ljy2855) [inline src/routers/captcha.py]: 이미 main에서 env설정되어있을테니 중복은 제거해주세요

File: src/routers/captcha.py [hunk 4]
```diff
@@ -1,108 +1,56 @@
...
-
     captcha_id = generate_captcha_id()
     expected = str(randint(0, 9))
     captcha_store[captcha_id] = expected
     return {"id": captcha_id, "expected": expected}
 
 
 @router.post("/predict")
-def predict(req: CaptchaRequest, request: Request):
+async def predict(req: CaptchaRequest, request: Request):
```
Comment (ljy2855) [inline src/routers/captcha.py:30]: 혹시 어떤 이유로, 비동기 처리로 추가하셨을까요?

File: src/saved_images/captcha_55b8927b-0f94-45a2-a39e-888ad1486bf9.png
Comment (ljy2855) [inline src/saved_images/captcha_55b8927b-0f94-45a2-a39e-888ad1486bf9.png:1]: saved images 폴더는 gitignore로 올려주심 감사하겠습니다 :)

File: src/main.py [hunk 5]
```diff
@@ -22,4 +29,67 @@
     max_age=3600,  # 세션 유지 시간 (초)
 )
 
+class InferenceRequest(BaseModel):
+    inputs: List[List[float]]
+
+@app.get("/")
```
Comment (ljy2855) [inline src/main.py]: main에 있는 endpoint를 router로 옮겨주셔요! `routers/model.y`

File: src/main.py [hunk 6]
```diff
@@ -22,4 +29,67 @@
     max_age=3600,  # 세션 유지 시간 (초)
 )
 
+class InferenceRequest(BaseModel):
```
Comment (ljy2855) [inline src/main.py]: 마찬가지로 Request model도 옮겨주셔요

File: src/routers/captcha.py [hunk 7]
```diff
@@ -1,108 +1,56 @@
...
+        # 원격 ML 서비스의 HybridCNN 모델 예측 API 호출
+        remote_url = f"{REMOTE_ML_SERVICE_URL}/models/predict/HybridCNN/"
+        async with httpx.AsyncClient() as client:
+            response = await client.post(remote_url, json=payload)
+            response.raise_for_status()
+            result = response.json()
         
+        predicted_digit = result["predictions"][0]
         passed = str(predicted_digit) == expected
+
```
Comment (ljy2855) [inline src/routers/captcha.py]: 혹시 `pre-commit install` 터미널에 한번 실행 가능할까요?

**Review by ljy2855 — CHANGES_REQUESTED** [review]
